### PR TITLE
#7494 Document Builder custom styles Slice 4: Update CustomFormRenderer brick input schema and config UI

### LINKED ISSUE #7494: Document Builder custom styles Slice 4: Update CustomFormRenderer brick input schema and config UI
[Slice spec here](https://www.notion.so/pixiebrix/Phase-3-Enable-custom-button-colors-in-the-Document-Builder-12a2a23b2e4a46158b53d0d04036a290?pvs=4#3abdd46ed36e4d0a86c03954b6042792)

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Closes #7494 
- Updates DocumentView so that custom stylesheets override bootstrap, if set
- document brick throws a BusinessError if there's an invalid stylesheet url

## Demo

image

## Checklist

- [ ] Add tests
- [ ] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer - @grahamlangford

### CODE REVIEW

File: src/bricks/renderers/document.ts [hunk 1]
```diff
@@ -92,18 +93,25 @@ export class DocumentRenderer extends RendererABC {
...
       stylesheets: string[];
     }>,
     options: BrickOptions,
   ): Promise<ComponentRef> {
+    const stylesheets = compact(_stylesheets);
+    for (const url of stylesheets) {
+      if (!isValidUrl(url)) {
+        throw new BusinessError(`Invalid Stylesheet URL: ${url}`);
+      }
+    }
```
Comment (BLoe) [inline src/bricks/renderers/document.ts:108]: @twschiller Here we're throwing an error now if there's an invalid stylesheet url

File: src/bricks/renderers/documentView/DocumentView.tsx [hunk 2]
```diff
@@ -43,21 +44,25 @@ const DocumentView: React.FC<DocumentViewProps> = ({
...
+    // in order to isolate the rendered document from the custom Bootstrap theme included in the Sidebar app
+    "/DocumentView.css",
+  ];
+
+  if (isEmpty(stylesheets)) {
+    stylesheetUrls.push(bootstrap, bootstrapOverrides);
+  } else {
+    // Custom stylesheets overrides bootstrap themes
+    stylesheetUrls.push(...stylesheets);
+  }
```
Comment (BLoe) [inline src/bricks/renderers/documentView/DocumentView.tsx:59]: @twschiller Custom stylesheets now override bootstrap, if set

File: src/pageEditor/fields/FormRendererOptions.tsx [hunk 3]
```diff
@@ -274,6 +269,13 @@ const FormRendererOptions: React.FC<{
           />
         </ConfigErrorBoundary>
       </ConnectedCollapsibleFieldSection>
+
+      <ConnectedCollapsibleFieldSection title={"Advanced: Theme"}>
+        <SchemaField
+          name={makeName("stylesheets")}
+          schema={CUSTOM_FORM_SCHEMA.properties.stylesheets as Schema}
+        />
+      </ConnectedCollapsibleFieldSection>
```
Comment (BLoe) [inline src/pageEditor/fields/FormRendererOptions.tsx:278]: This is the new input on the custom form config UI that is the main goal of this slice

File: src/bricks/renderers/customForm.ts [hunk 4]
```diff
@@ -169,24 +169,33 @@ export const customFormRendererSchema = {
...
     },
+    stylesheets: {
+      type: "array",
+      items: {
+        type: "string",
+      },
+      title: "CSS Stylesheet URLs",
+      description:
+        "Stylesheets will apply to the rendered document in the order listed here",
+    },
```
Comment (BLoe) [inline src/bricks/renderers/customForm.ts:180]: This is the main schema change for this slice

File: src/bricks/renderers/document.ts (hunk 1 above)
Comment (twschiller) [reply src/bricks/renderers/document.ts:108]: NIT: prefer PropError for errors about a particular prop

Remind me why we aren't setting the format to uri for the stylesheet items? Those would then be checked prior to the brick being called

**Review by twschiller — APPROVED** [review]

File: src/bricks/renderers/document.ts (hunk 1 above)
Comment (BLoe) [reply src/bricks/renderers/document.ts:108]: Sorry, missed this, will address in the next PR

**Pull request merged**